Lib/inspect.py

```diff
@@ -3107,6 +3128,17 @@ def from_callable(cls, obj, *,
                                         follow_wrapper_chains=follow_wrapped,
                                         globals=globals, locals=locals, eval_str=eval_str)
 
+    @classmethod
+    def from_code(cls, co):
+        """Constructs Signature for the given code object.
+
+        Signatures created from code objects cannot know about annotations
+        or default values.
+        """
+        if not iscode(co):
+            raise TypeError(f'code object is expected, got {type(co)!r}')
```

Comment:
The usual style for Python `TypeError`s is to quote the _name_ of the incorrect type rather than the _repr_ of the incorrect type:

```pycon
>>> 'foo' + 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate str (not "int") to str
```

(also, we usually use "was" rather than "is")

```suggestion
            raise TypeError(f'code object was expected, got {type(co).__name__!r}')
```